### PR TITLE
[fix] support and pin undici 5.12.0

### LINKED ISSUE #7409: +server.ts API endpoint fails to parse body
### Describe the bug

The only file new file in the blank svelte-kit project is the `+server.ts` located in `src/routes/api/+server.ts`. It just echoes the incoming JSON. Hitting `http://localhost:5173/api` (with `content-type` and `accept` headers set to `application/json` and body `{ "sample": "test"}` ) will return `HTTP 400 Bad request` -> and the request never hits the API endpoint.

If I remove the content-type and accept headers it will enter the API endpoint but it would error out on `await request.json()` with:

```
Unexpected end of JSON input
SyntaxError: Unexpected end of JSON input
```

It might be a Windows issue but honestly I am not sure and can't test atm.

### Reproduction

https://github.com/callmeberzerker/body-not-working

Run `pnpm install` and then `pnpm dev`.

### Logs

```Shell
Sadly there are none.
```

### System Info

```Shell
System:
    OS: Windows 10 10.0.22621
    CPU: (16) x64 Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz
    Memory: 16.30 GB / 31.92 GB
  Binaries:
    Node: 19.0.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - C:\Program Files\nodejs\yarn.CMD
    npm: 8.19.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.674.0), Chromium (105.0.1343.42)
    Internet Explorer: 11.0.22621.1
  npmPackages:
    vite: ^3.2.0 => 3.2.0
```

### Severity

blocking all usage of SvelteKit

### Additional Information

N/A

### PR DESCRIPTION
**Pull request by repsac-by**
Fixes #7409 

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

**Comment by jalavosus** [comment]
Not to knock your fix (it's a good one) but wouldn't a more stable fix be to pin to the last known good version of undici (5.11.0)?

**Comment by repsac-by** [comment]
There are many fixes and improvements in 5.12.0.
I don't see any good reason to stay with 5.11.0 other than the addition of the `duplex` property.

**Comment by N00nDay** [comment]
Worked like a charm for me. Thanks repsac-by

**Comment by benmccann** [comment]
The Undici team indicated this is the correct fix

>This is part of the spec now. When you have a ReadableStream body, the duplex option is required.

File: packages/kit/src/exports/node/index.js [hunk 1]
```diff
@@ -95,6 +95,8 @@ function get_raw_body(req, body_size_limit) {
 /** @type {import('@sveltejs/kit/node').getRequest} */
 export async function getRequest({ request, base, bodySizeLimit }) {
 	return new Request(base + request.url, {
+		// @ts-expect-error
```
Comment (benmccann) [inline packages/kit/src/exports/node/index.js:98]: I would expect we shouldn't need this. Is there some detail we can add here about how/when we might be able to remove it. Are the types provided by Typescript or Undici? Is there and open issue we can link to or can we file a new one?

**Comment by Conduitry** [comment]
The undici PR included updates to the types here (https://github.com/nodejs/undici/pull/1681/files#diff-cfe294cd3fb885afcce564fae32dd1524399460a9a04f141e07bb45858a47f92), but it looks like we're not using those. I tries this out locally (and updated the Kit repo to undici 5.12.0), but the types for `Request` appeared to be coming from TypeScript (`lib.dom.d.ts`) and not from undici. This `@ts-expect-error` is probably the best we can do for now. Once the types are added to TS and we upgrade, we'll start getting an error here, so at least we'll remember to remove this.

This change looks good to me. Can we get a changeset?

File: packages/kit/src/exports/node/index.js (hunk 1 above)
Comment (Conduitry) [reply packages/kit/src/exports/node/index.js:98]: These types appear to be coming from TS, not from undici. Locally, even when I updated the Kit repo to undici 5.12.0, TS still complained about the `duplex`.

**Comment by benmccann** [comment]
We should probably pin the Undici version as well as they've indicated that it's experimental and may have breaking changes in minor/patch versions: https://github.com/nodejs/undici/pull/1681#issuecomment-1294178466

**Comment by Conduitry** [comment]
That sounds good to me as well.

**Review by benmccann — APPROVED** [review]

**Pull request merged**